Passes/PrototypeDeferredPipeline/DeferredLighting.azsl

```diff
@@ -0,0 +1,160 @@
+{
+    uint2 screenCoords = IN.m_position.xy;
+
+    // ------- Unpack G-Buffers -------
+
+    float zDepth = PassSrg::m_depthStencilTexture.Load(screenCoords, sampleIndex).r;
+    float3 surfacePosWS = WorldPositionFromDepthBufferMS(PassSrg::m_depthStencilTexture, zDepth, IN.m_position.xy).xyz;
+    float3 encodedNormal = PassSrg::m_normal.Load(screenCoords, sampleIndex).xyz;
+    float3 normal = DecodeNormalSignedOctahedron(encodedNormal.xyz);
+    float2 roughnessMetallic = PassSrg::m_roughnessMetal.Load(screenCoords, sampleIndex).rg;
+    float3 baseColor = PassSrg::m_baseColor.Load(screenCoords, sampleIndex).rgb;
+    
+    if(o_deferredPipelineWatermark)
+    {
+        if(abs(IN.m_texCoord.x-0.5) < 0.01 || abs(IN.m_texCoord.y-0.5) < 0.01) 
+        {
+            float t = sin(50 * (IN.m_texCoord.x + IN.m_texCoord.y)) * 0.5 + 0.5;
+            baseColor *= lerp(float3(1,0.2,1), float3(0.5,1,1), t); 
+        }
+    }
+
+    // Copied from Gems\Atom\Feature\Common\Assets\Materials\Pipelines\MainPipeline\ForwardPass.azsli
+
+    // ------- Surface -------
+
+    Surface surface;
+    surface.position = surfacePosWS;
+
+    // TODO(DeferredPOC): We don't actually have the vertex normal, so just use the per-pixel normal for now
+    surface.vertexNormal = normal;
+    surface.normal = normal;
+    surface.roughnessLinear = roughnessMetallic.x;
+    float specularF0Factor = 0.5f; // TODO(DeferredPOC): Consider passing the actual specF0 through a GBuffer
+    surface.SetAlbedoAndSpecularF0(baseColor, specularF0Factor, roughnessMetallic.y);
+    surface.clearCoat.InitializeToZero();
+        
+    surface.CalculateRoughnessA();
+
+    // ------- LightingData -------
+
+    LightingData lightingData;
+
+    // Light iterator
+    lightingData.tileIterator.Init(IN.m_position, PassSrg::m_lightListRemapped, PassSrg::m_tileLightData);
+    lightingData.Init(surface.position, surface.normal, surface.roughnessLinear);
+
+
+    // Diffuse and Specular response
+    lightingData.specularResponse = FresnelSchlickWithRoughness(lightingData.NdotV, surface.specularF0, surface.roughnessLinear);
+    lightingData.diffuseResponse = 1.0f - lightingData.specularResponse;
+
+    const float alpha = 1.0f;
+
+    // ------- Lighting Calculation -------
+
+    // Apply Decals
+    ApplyDecals(lightingData.tileIterator, surface);
+
+    // Apply Direct Lighting
+    ApplyDirectLighting(surface, lightingData, IN.m_position);
+
+    // Apply Image Based Lighting (IBL)
+    // TODO(DeferredPOC): how can we support a localized reflection probe in deferred?
+    ReflectionProbeData unusedReflectionProbe; 
+    TextureCube unusedReflectionProbeCubeMap;
+    ApplyIBL(surface, lightingData, true, false, unusedReflectionProbe, unusedReflectionProbeCubeMap);
+
+    // Finalize Lighting
+    lightingData.FinalizeLighting();
+
+    PbrLightingOutput lightingOutput = GetPbrLightingOutput(surface, lightingData, alpha);
```

Comment:
Can just call EvaluateLighting_StandardPBR, no need to duplicate the code, or does this do something different?

Reply:
Yes, I believe I made this file before the big shader refactor, and I can come back and refactor this code later.